packages/web-app-files/tests/unit/components/SideBar/Versions/FileVersions.spec.js

```diff
@@ -0,0 +1,265 @@
+import Vuex from 'vuex'
+import { DateTime } from 'luxon'
+import stubs from '@/tests/unit/stubs'
+import GetTextPlugin from 'vue-gettext'
+import DesignSystem from 'owncloud-design-system'
+import { mount, shallowMount, createLocalVue } from '@vue/test-utils'
+import FileVersions from '@files/src/components/SideBar/Versions/FileVersions.vue'
+
+const defaultStubs = {
+  ...stubs,
+  'oc-td': true,
+  'oc-tr': true,
+  'oc-tbody': true,
+  'oc-table-simple': true
+}
+
+const yesterday = DateTime.now()
+  .minus({ days: 1 })
+  .toHTTP()
+
+const sevenDaysBefore = DateTime.now()
+  .minus({ days: 7 })
+  .toHTTP()
+
+const localVue = createLocalVue()
+localVue.use(Vuex)
+localVue.use(DesignSystem)
+localVue.use(GetTextPlugin, {
+  translations: 'does-not-matter.json',
+  silent: true
+})
+
+const defaultVersions = [
+  {
+    fileInfo: {
+      '{DAV:}getcontentlength': '23',
+      '{DAV:}getcontenttype': 'text/plain',
+      '{DAV:}getetag': '"82add182994ade91e3d5bc47571ea731"',
+      '{DAV:}getlastmodified': yesterday,
+      '{DAV:}resourcetype': ''
+    },
+    name: '/meta/2147524174/v/1625818937',
+    tusSupport: null,
+    type: 'file'
+  },
+  {
+    fileInfo: {
+      '{DAV:}getcontentlength': '11',
+      '{DAV:}getcontenttype': 'text/plain',
+      '{DAV:}getetag': '"311b3319ebc7063069a15ee02b926298"',
+      '{DAV:}getlastmodified': sevenDaysBefore,
+      '{DAV:}resourcetype': ''
+    },
+    name: '/meta/2147524174/v/1625637401',
+    tusSupport: null,
+    type: 'file'
+  }
+]
+
+const selectors = {
+  loadingStub: 'oc-loader-stub',
+  versionTableStub: 'oc-simple-table-stub',
+  noVersionsMessage: 'span',
+  fileTypeIcon: '.file-icon oc-icon-stub',
+  lastModifiedDate: '.file-last-modified-date',
+  resourceSize: '.file-size',
+  revertVersionButton: '.oc-files-revert-version-button',
+  downloadVersionButton: '.oc-files-download-version-button'
+}
+
+const mapActions = {
+  loadVersions: jest.fn()
+}
+
+function getShallowWrapper(store) {
+  return shallowMount(FileVersions, {
+    localVue,
+    store: store,
+    stubs: defaultStubs,
+    directives: {
+      'oc-tooltip': true
+    }
+  })
+}
```

Comment:
you can remove `getLoadingWrapper` if you edit `getShallowWrapper` like below

and use like `const wrapper = getLoadingWrapper(createStore(), true)` to test loading componnent
```suggestion
function getShallowWrapper(store, loading = false) {
  return shallowMount(FileVersions, {
    localVue,
    store: store,
    stubs: defaultStubs,
    directives: {
      'oc-tooltip': true
    },
    data() {
      return {
        loading
      }
    }
  })
}
```